src/Symfony/Component/Debug/DebugClassLoader.php

```diff
@@ -228,40 +183,88 @@ public function loadClass($class)
                 }
             }
 
-            foreach (array_merge(array($parent), class_implements($name, false), class_uses($name, false)) as $use) {
+            // Detect annotations on the class
+            if (false !== $doc = $refl->getDocComment()) {
+                foreach (array('final', 'deprecated', 'internal') as $annotation) {
+                    if (false !== strpos($doc, '@'.$annotation) && preg_match('#\n \* @'.$annotation.'(?:( .+?)\.?)?\r?\n \*(?: @|/$)#s', $doc, $notice)) {
+                        self::${$annotation}[$name] = isset($notice[1]) ? preg_replace('#\s*\r?\n \* +#', ' ', $notice[1]) : '';
+                    }
+                }
+            }
+
+            if ($parent = get_parent_class($class)) {
+                if (isset(self::$final[$parent])) {
+                    @trigger_error(sprintf('The "%s" class is considered final%s. It may change without further notice as of its next major version. You should not extend it from "%s".', $parent, self::$final[$parent], $name), E_USER_DEPRECATED);
+                }
+            }
+
+            $traits = class_uses($name, false);
```

Comment:
`$traits = array($parent) + class_uses($name, false);`
Then below `foreach ($interfaces + $traits` to remove the array_merge() (or directly $traits, maybe renamed btw)